java-checks/src/main/java/org/sonar/java/checks/CompareStringsBoxedTypesWithEqualsCheck.java

```diff
@@ -19,29 +19,107 @@
+    Optional<JavaQuickFix> conciseQuickFix = computeConciseQuickFix(tree);
+    if (conciseQuickFix.isPresent()) {
+      return Collections.singletonList(conciseQuickFix.get());
+    }
+    return Collections.singletonList(computeDefaultQuickFix(tree));
+  }
+
+  private static Optional<JavaQuickFix> computeConciseQuickFix(BinaryExpressionTree tree) {
+    if (tree.leftOperand().is(Tree.Kind.STRING_LITERAL)) {
+      AnalyzerMessage.TextSpan interOperandSpace = AnalyzerMessage.textSpanBetween(
+        tree.leftOperand().lastToken(), false,
+        tree.rightOperand().firstToken(), false
+      );
+      List<JavaTextEdit> edits = new ArrayList<>();
+      edits.add(JavaTextEdit.insertAfterTree(tree.rightOperand().lastToken(), ")"));
+      edits.add(JavaTextEdit.replaceTextSpan(interOperandSpace, DOT_EQUALS_AND_OPENING_PARENTHESIS));
+      if (tree.is(Tree.Kind.NOT_EQUAL_TO)) {
+        edits.add(JavaTextEdit.insertBeforeTree(tree.leftOperand().firstToken(), "!"));
+      }
+      JavaQuickFix.Builder quickFix = JavaQuickFix.newQuickFix(QUICK_FIX_MESSAGE)
+        .addTextEdit(JavaTextEdit.insertAfterTree(tree.rightOperand().lastToken(), ")"))
+        .addTextEdit(JavaTextEdit.replaceTextSpan(interOperandSpace, DOT_EQUALS_AND_OPENING_PARENTHESIS));
+      if (tree.is(Tree.Kind.NOT_EQUAL_TO)) {
+        quickFix.addTextEdit(JavaTextEdit.insertBeforeTree(tree.leftOperand().firstToken(), "!"));
+      }
+      return Optional.of(quickFix.build());
+    } else if (tree.rightOperand().is(Tree.Kind.STRING_LITERAL)) {
+      String callEqualsOnLiteral = ((LiteralTree) tree.rightOperand()).value() + DOT_EQUALS_AND_OPENING_PARENTHESIS;
+      String callToEquals = tree.is(Tree.Kind.NOT_EQUAL_TO) ? ("!" + callEqualsOnLiteral) : callEqualsOnLiteral;
+      AnalyzerMessage.TextSpan leftOfOperatorToEndOfComparison = AnalyzerMessage.textSpanBetween(
+        tree.leftOperand().lastToken(), false,
+        tree.rightOperand().lastToken(), true
+      );
+      return Optional.of(
+        JavaQuickFix.newQuickFix(QUICK_FIX_MESSAGE)
+          .addTextEdit(JavaTextEdit.removeTextSpan(leftOfOperatorToEndOfComparison))
+          .addTextEdit(JavaTextEdit.insertAfterTree(tree.leftOperand().lastToken(), ")"))
+          .addTextEdit(JavaTextEdit.insertBeforeTree(tree.leftOperand().firstToken(), callToEquals))
+          .build()
+      );
+    }
+    return Optional.empty();
+  }
+
+  private static JavaQuickFix computeDefaultQuickFix(BinaryExpressionTree tree) {
+    String callToEquals = tree.is(Tree.Kind.NOT_EQUAL_TO) ? "!java.util.Objects.equals(" : "java.util.Objects.equals(";
+    AnalyzerMessage.TextSpan interOperandSpace = AnalyzerMessage.textSpanBetween(
+      tree.leftOperand().lastToken(), false,
+      tree.rightOperand().firstToken(), false
+    );
+    return JavaQuickFix.newQuickFix(QUICK_FIX_MESSAGE)
+      .addTextEdit(JavaTextEdit.insertAfterTree(tree.rightOperand().lastToken(), ")"))
```

Comment:
```suggestion
      .addTextEdit(JavaTextEdit.insertAfterTree(tree.rightOperand(), ")"))
```
Inserting after the rightOperand or after the last token is the same, you can remove the call. (In practice, the method anyway takes the last token.)

Reply:
And in fact, you can nearly always omit the call to last/first token, even when calling `textSpanBetween`.